extensions/basic_auth/README.md

```diff
@@ -1,59 +1,33 @@
-
-Then apply another `EnvoyFilter` to inject basic auth filter into HTTP filter chain. For example, [this configuration](./config/gateway-filter.yaml) injects the basic auth filter to `gateway`. Most of the configuration is boilerplate, the important parts are:
-
-* The `EnvoyFilter` matches gateway proxy with version 1.8, since the module downloaded is of version 1.8.
-  ```yaml
-  - applyTo: HTTP_FILTER
-    match:
-      context: GATEWAY
-      listener:
-        filterChain:
-          filter:
-            name: envoy.http_connection_manager
-      proxy:
-        proxyVersion: ^1\.8.*
-  ```
-* Basic auth rules, which configures basic auth filter to do auth check based on prefix, HTTP method, and the given credentials.
-  ```json
-  {
-    "basic_auth_rules": [
-      {
-        "prefix": "/",
-        "request_methods": ["GET", "POST"],
-        "credentials": [
-          "ok:test",
-          "admin:admin",
-          "admin2:admin2"
-        ]
-      }
-    ]
-  }
-  ```
-* Module downloading URL and checksum. To get the checksum, you can use `sha256sum` command: `sha256sum ${YOUR_WASM_MODULE}`.
-  ```yaml
-  remote:
-    http_uri:
-      uri: https://storage.googleapis.com/istio-ecosystem/wasm-extensions/basic-auth/1.8.0.wasm
-      cluster: google-storage
-      timeout: 10s
-    sha256: 707e29db817f76c974d7ce1fe2f61ad64c88856c7ddba99a36fe95439bfe1281
-  ```
-
-After applying the filter, gateway should start enforce the basic auth rule. Use `productpage` app as an example, to test that the rule works, you can curl with and without the authorization header. For example
+In the following guide we will configure Istio proxy to download and apply Basic Auth filter.
+
+Two `EnvoyFilter` resources will be applied, to inject basic auth filter into HTTP filter chain.
+For example, [this configuration](./config/gateway-filter.yaml) injects the basic auth filter to `gateway`.
+
+The first `EnvoyFilter` will be inject an HTTP filter into gateway proxies.
+It is configured to request the extension configuration named as `istio.basic_auth` from `ads` (i.e. Aggregated Discovery Service), which is the same configuration source that Istiod uses to provide all other configuration resources.
+Along with the configuration source, the initial fetch timeout is also set, in order to prevent filters with slow or failed Wasm module remote fetch becomes effective.
+The second `EnvoyFilter` resource provides configuration for the filter, which is composed as an `EXTENSION_CONFIG` patch and will be distributed to the proxy as an Envoy [`Extension Configuration`](https://www.envoyproxy.io/docs/envoy/latest/configuration/overview/extension) (ECDS) resource.
+Once this update reaches Istio agent, it will download the Wasm module file and store it at the local file system.
+If the download fails, Istio-agent will reject the `Extension Configuration` update and prevent bad Wasm filter configuration from reaching Envoy.
```

Comment:
do we have a stats metric for ECDS update failure? if so can we add the name here as well so users can monitor

Reply:
Yeah it is already documented on istio.io, and there is actually lots of duplication here to that document: https://istio.io/latest/docs/ops/configuration/extensibility/wasm-module-distribution/

I removed bunch of content from the readme guide and redirect user to read the official istio.io guide instead.